recipe-server/client/control_new/components/common/EnvAlert.js

```diff
@@ -0,0 +1,66 @@
+import { Alert } from 'antd';
+import React from 'react';
+
+/**
+ * Simple component which tells the user whether they are viewing a dev or staging
+ * environment. On production, nothing is shown.
+ */
+export default class EnvAlert extends React.PureComponent {
+  static productionUrls = ['normandy-admin.prod.'];
+  static stageUrls = ['normandy-admin.stage.'];
+
+  static learnMoreLink = (
+    <a
+      href="http://normandy.readthedocs.io/en/latest/qa/environments.html"
+      target="_blank"
+      rel="noopener noreferrer"
+    >
+      Learn more about Normandy environments.
+    </a>);
+
+  /**
+   * Find a string fragment in an array of strings.
+   *
+   * @param  {String}        needle   String fragment to find.
+   * @param  {Array<String>} haystack String collection to search.
+   * @return {Boolean}                Was the fragment found?
+   */
+  static findPartialString(needle, haystack) {
+    return !!haystack.find(straw => straw.indexOf(needle) > -1);
```

Comment:
I think this doesn't work right. The `needle` will be the the full url, `https://normandy-admin.stage.mozaws.net/control/...`, which won't be a substring of any of the `haystack`.